### ISSUE TITLE
Optional properties on IExponentialBackoffOptions should be marked as optional

### ISSUE THREAD

**Issue by xzyfer** [connor4312/cockatiel#6]
Currently trying to do the following

```js
.exponential({ maxDelay: 1000, initialDelay: 10, maxAttempts: 2 })
```

Results in the following error

```
Argument of type '{ maxDelay: number; initialDelay: number; maxAttempts: number; }' is not assignable to parameter of type 'IExponentialBackoffOptions<IRetryBackoffContext<unknown>>'.
  Type '{ maxDelay: number; initialDelay: number; maxAttempts: number; }' is missing the following properties from type 'IExponentialBackoffOptions<IRetryBackoffContext<unknown>>': generator, exponent
```

However the `ExponentialBackoff` constructor merges the supplied options with the a default set of options. This means all properties on `IExponentialBackoffOptions` are optional.

**Comment by xzyfer** [comment]
Additionally when trying to satisfy the interface with the `defaultOptions` from `ExponentialBackoff`

```
.exponential({
    generator: decorrelatedJitterGenerator,
    maxDelay: 30000,
    maxAttempts: Infinity,
    exponent: 2,
    initialDelay: 128,
});
```

Results in the following error

```
Type 'GeneratorFn<[number, number]>' is not assignable to type 'GeneratorFn<IRetryBackoffContext<unknown>>'.
  Type '[number, number]' is not assignable to type 'IRetryBackoffContext<unknown>'.
```

**Comment by connor4312** [comment]
Thanks, fixed in f96beb5 and published in 0.1.2.
